### PR TITLE
Make client creation with self-issed tokens easier. Fixes #9

### LINKED ISSUE #9: Document usage or extend API for self-issued access tokens
It's possible to use the package with an AccessToken only. 

```
        m := &medium.Medium{
                AccessToken: "TOKEN",
                Host:        "https://api.medium.com",
        }
```

I think this should be documented or better yet API extended for this use case.

### PR DESCRIPTION
**Pull request by majelbstoat**
Hello @dpup, @vinibaggio, 

Please review the following commits I made in branch 'jamie-self-issued'.

0dc3042aae4c751351572f110b418aa72efe9fe0 (2015-10-08 13:31:45 -0700)
Make client creation with self-issed tokens easier. Fixes #9

Thoughts on naming, @dpup?

R=@dpup
R=@vinibaggio

MANUAL TESTING=not manually tested

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,6 +41,10 @@ func main() {
 	// directly.
 	m.AccessToken = at.AccessToken
 
+  // If you have a self-issued access token, you can skip these steps and
+  // create a new client directly:
+  m = medium.NewClientWithAccessToken("SELF_ISSUED_ACCESS_TOKEN")
```
Comment (vinibaggio) [inline README.md]: nit: indentation seems off?

**Comment by vinibaggio** [comment]
LGTM

File: README.md (hunk 1 above)
Comment (dpup) [inline README.md]: Maybe `m2 :=` to disambiguate it in the example? Or a different main()

**Comment by dpup** [comment]
LGTM

**Pull request merged**